### PR TITLE
Run Tests on PR Create/Update

### PR DESCRIPTION
**Pull request by captainrandom**
# Motivation and context
This runs unit tests when a pull request is created and when a pull request is updated.

Something I probably should do is make the local python setup file use the requirements file as we are starting to get one too many setup scripts for python. (really just need one that works across the board)

# Screenshots
Here is a url of the build working in my repo:
https://github.com/captainrandom/public-tree-map-data-pipeline/runs/624873917?check_suite_focus=true

# What I did
- added a github action workflow to run python tests on pr create/update

Keep in mind that only my script has unit tests being run on it. As other code is touched/added, we will add more unit tests to make sure that stuff is working.

### CODE REVIEW

**Comment by captainrandom** [comment]
You are going to have to enable the tests as part of the branch merge requirements. I can walk you through this next week.

**Comment by emillipede** [comment]
thank you @captainrandom !

> Something I probably should do is make the local python setup file use the requirements file as we are starting to get one too many setup scripts for python. (really just need one that works across the board)

which requirements file do you mean? I agree about setup scripts. is it, for you, another argument to use containers?

File: setup_python_locally.sh [hunk 1]
```diff
@@ -4,4 +4,5 @@ conda create -n tree-map
 conda activate tree-map
 conda config --env --add channels conda-forge
 conda config --env --set channel_priority strict
-conda install python=3 geopandas==0.4.0
\ No newline at end of file
+conda install python=3
+conda install --file requirements.txt
```
Comment (captainrandom) [inline setup_python_locally.sh:8]: @emillipede, in reference to your earlier question regarding setup scripts being in multiple places. This is one such location (the setup_python_locally.sh script). The other one lives in the python-env.sh, which references the environment.yml script:
https://github.com/Public-Tree-Map/public-tree-map-data-pipeline/blob/master/environment.yml

Comment (captainrandom) [reply setup_python_locally.sh:8]: It's not directly an argument for containers. It's more like having one setup script that runs locally and circleci.

**Comment by captainrandom** [comment]
Screen Shot 2020-05-04 at 7 26 10 PM

**Comment by captainrandom** [comment]
Screen Shot 2020-05-04 at 7 27 29 PM

**Pull request merged**